### ISSUE TITLE
Create scenario to run agents on a Fairground market 

### ISSUE THREAD

**Issue by cdummett** [vegaprotocol/vega-market-sim#178]
### Description

Recent PRs have added the capability to run `vega-market-sim` agents on any Vega network. This ticket is for creating a scenario which can be deployed by DevOps on a market setup specifically for running `vega-market-sim` agents on. The scenario requires the following features:

1. `IdealMarketMaker` agent (provides a market)
2. `SimpleLiquidityProvider` agents (provide enough liquidity for the expected open  interest of the traders)
3. `MomentumTrader` agents (possibly multiples for each strategy with a variety of strategy parameters).
4. `BackgroundMarket` which follows a historic price-process (ideally a live price-process but historic for now).
5. `MarketOrderAgents` which guarantee trades each step to ensure candles are created for the `MomentumTraders`

Some of the above agents will need some modifications to be able to run with multi-key wallets.

### Extras

As an additional nice to have it should be possible to run the same scenario on a nullchain network. This would allow testing of network parameters which can be given to DevOps when setting up the market. For this the scenario would require the following extra features:

1. `MarketManager` agent which can propose and settle a market (should be configurable so as to not run on fairground).
2. `adhoc.py` modifications to allow network selection

Some more mods might need to be made to the `SlimWallet` to allow multi-key wallets. Original mods were made with the intention the `SlimWallet` would only ever single-key wallets.

**Comment by davidsiska-vega** [comment]
I'd just try to make it possible to run e.g. `vega_sim/scenario/curve_market_maker/scenario.py` so: 

```
scenario = CurveMarketMaker(
        market_decimal=3,
        asset_decimal=5,
        market_position_decimal=2,
        initial_price=1000.0,
        lp_commitamount=100000,
        initial_asset_mint=1e8,
        step_length_seconds=1,
        block_length_seconds=1,
        buy_intensity=5,
        sell_intensity=5,
        state_extraction_fn=state_fn,
        market_name=market_name,
        num_steps=2,
        random_agent_ordering=False,
        sigma=100,
    )
    env = scenario.set_up_background_market(
        vega=vega,
        tag=str(step_tag),
    )

    result = env.run(
        run_with_console=run_with_console,
        pause_at_completion=pause_at_completion,
    )
```
but with the necessary changes. Or replicate this scenario as much as possible for the purpose of running on a vega network if it's easier. Probably best to check with @TomMcL  really.

**Comment by cdummett** [comment]
Sounds good thanks for the feedback, what I have so far follows the scenario of that structure.

The main thing I've changed is having the `ExponentialShapedMarketMaker` follow a "random" historic price process rather than a random walk so we can deploy some `MomentumTrader` agents on the market and hopefully see some wins.

Turns out I do need to make a few changes to the `SlimWallet` and `StateAgentWithWallet` parent class as well to get mutli-key agents working on nullchain (e.g. change `create_wallet` behaviour to handle a wallet which already exists etc...)
